### DISCUSSION TITLE
Can document_exporter export to some kind of hierarchical folders?

### DISCUSSION THREAD

**Discussion by entropical** [jonaswinkler/paperless-ng#1426] [Feature Requests]
I setup document_exporter, and then rclone to onedrive. All good, no problems.

Except now I have 1000s of files in a single directory, which can be a pain (under Windows specifically).

I thought it might be handy to allow exports to be automatically separated into some kind of (selectable) directory structure.

- yyyy/mm/dd
- \<correspondent\>/\<document_type\>/yyyy

If this seems like a reasonable idea, I can look at implementing it.

**Comment by Conspir3D** [comment]
dosn't this help?

use this: --use-filename-format

documented at: https://paperless-ng.readthedocs.io/en/latest/administration.html?highlight=export#document-exporter

**Comment by entropical** [comment]
Maybe, although it doesn't explicitly say if that allows directory separators. I'll give it a go later.

**Comment by tarioch** [comment]
works very well, that's how I have set things up:

```
{document_type}/{created_year}/{created_month}-{created_day}_{correspondent}_{title}
```
